src/target/arc_mem.c

```diff
@@ -76,7 +74,10 @@ static int arc_mem_write_block32(struct target *target, uint32_t addr,
 	/* Check arguments */
 	assert(!(addr & 3));
 
-	/* No need to flush cache, because we don't read values from memory. */
+	/* We need to flush the cache since it might contain dirty
+	 * lines, so the cache invalidation may cause data inconsistency. */
+	CHECK_RETVAL(arc32_cache_flush(target));
```

Comment:
You also need to update comment in write block 16 and 8, because reason why they need to flush cache is now different.

Also change commit title - it should reflect what type of bug you are fixing. It can just reflect that you change something, and then in the description you explain that there was a bug. So title should be something like "arc: Flush cache before writing memory".